### PR TITLE
fix(PTR-24616798): remove default basePath for api clients

### PR DESCRIPTION
**Pull request by vscaiceanu-1a**

### CODE REVIEW

File: packages/@ama-sdk/core/src/fwk/api.interface.ts [hunk 1]
```diff
@@ -16,3 +16,5 @@ export interface Api extends ApiName {
 export interface ApiConstructor<T extends Api = Api> extends ApiName {
   new (client: ApiClient): T;
 }
+
+export type PartialExcept<T, K extends keyof T> = Partial<T> & Pick<T, K>;
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/api.interface.ts:20]: missing comments

File: packages/@ama-sdk/core/src/fwk/clients/api-beacon-client.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import type { RequestBody, RequestMetadata, RequestOptions, TokenizedOptions } from '../../plugins';
 import type { ApiTypes } from '../api';
 import { extractQueryParams, filterUndefinedValues, prepareUrl, processFormData, tokenizeRequestOptions } from '../api.helpers';
+import { PartialExcept } from '../api.interface';
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/clients/api-beacon-client.ts:4]: could be `import type`

File: packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { ExceptionReply } from '../../plugins/exception';
 import { ReviverReply } from '../../plugins/reviver';
 import { ApiTypes } from '../api';
 import { extractQueryParams, filterUndefinedValues, prepareUrl, processFormData, tokenizeRequestOptions } from '../api.helpers';
+import { PartialExcept } from '../api.interface';
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts:15]: could be `import type`

File: packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts [hunk 4]
```diff
@@ -23,14 +24,16 @@ export interface BaseApiFetchClientOptions extends BaseApiClientOptions {
   fetchPlugins: FetchPlugin[];
 }
 
+
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts:27]: too many empty lines

File: packages/@ama-sdk/core/src/plugins/gateway-token/gateway-authentication.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable no-redeclare */
 /* eslint-disable camelcase, @typescript-eslint/naming-convention */
 
-import { Api, ApiClient, ApiFetchClient, ApiTypes } from '../../fwk/index';
+import { Api, ApiClient, ApiTypes } from '../../fwk/index';
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/plugins/gateway-token/gateway-authentication.ts:4]: could be `import type` ?

File: packages/@ama-sdk/core/src/fwk/api.interface.ts [hunk 6]
```diff
@@ -16,3 +16,5 @@ export interface Api extends ApiName {
 export interface ApiConstructor<T extends Api = Api> extends ApiName {
   new (client: ApiClient): T;
 }
+/** Type that is the same as the `Partial<T>` type, but with a certain property excluded. */
```
Comment (cpaulve-1A) [inline packages/@ama-sdk/core/src/fwk/api.interface.ts:19]: comment is unclear, the way I understand it is that the property K would not be part of the type while you have a mandatory K.

File: packages/@ama-sdk/core/src/fwk/clients/api-beacon-client.ts [hunk 7]
```diff
@@ -11,13 +12,12 @@ export interface BaseApiBeaconClientOptions extends BaseApiClientOptions {
 }
 
 /** @see BaseApiConstructor */
-export interface BaseApiBeaconClientConstructor extends Partial<Omit<BaseApiBeaconClientOptions, 'replyPlugins'>> {
+export interface BaseApiBeaconClientConstructor extends PartialExcept<Omit<BaseApiBeaconClientOptions, 'replyPlugins'>, 'basePath'> {
+  /** API Gateway base path (when targeting a proxy or middleware) */
+  basePath: string;
```
Comment (cpaulve-1A) [inline packages/@ama-sdk/core/src/fwk/clients/api-beacon-client.ts:17]: is this one necessary since it is part of your PartialExcept type?

**Review by cpaulve-1A — DISMISSED** [review]

File: packages/@ama-sdk/core/src/fwk/clients/api-beacon-client.ts [hunk 8]
```diff
@@ -37,25 +35,23 @@ const isPromise = <T>(value: T | Promise<T>): value is Promise<T> => value && ty
 export class ApiBeaconClient implements ApiClient {
 
   /** @inheritdoc */
-  public options = DEFAULT_OPTIONS;
+  public options;
```
Comment (mrednic-1A) [inline packages/@ama-sdk/core/src/fwk/clients/api-beacon-client.ts:38]: the type should be BaseApiBeacontClientOPtions ? Maybe it is not needed if we implement an interface :)

**Review by mrednic-1A — DISMISSED** [review]

**Review by mrednic-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**